ivy/functional/backends/paddle/elementwise.py

```diff
@@ -125,6 +123,10 @@ def equal(
     )
 
 
+@with_supported_dtypes(
+    {"2.5.1 and below": ("bool", "float32", "float64", "int32", "int64", "complex")},
```

Comment:
wouldn't it be easier to mark `int8` and `uint8` as unsupported instead?

Reply:
I'm not sure, If paddle actually supports them, shouldn't we prevent any unnecessary type casting ?

Reply:
Sneaking in, haha, the role of the decorators is to specify the dtypes not supported by a given function but supported by the backend itself : in the case of `with_unsupported_dtypes`

In the case of `with_supported_dtypes`: the role of the decorators is to specify the ****only**** dtypes supported by a function.

So in this case, since the function doesn't support `int8` and `uint8` we provide that information through the `unsupported` dtype decorators. We don't specify information about the dtypes not supported by the backend at all because this information is redundant and is known already.

Reply:
> I'm not sure, If paddle actually supports them, shouldn't we prevent any unnecessary type casting ?

What I meant was that if the number of unsupported dtypes is less than the number of supported dtypes, it might be more convenient to use `with_unsupported_dtypes` instead of `with_supported_dtypes`. Either way would work, but just was curious about which ones more convenient.

Reply:
> Sneaking in, haha, the role of the decorators is to specify the dtypes not supported by a given function but supported by the backend itself : in the case of `with_unsupported_dtypes`
> 
> In the case of `with_supported_dtypes`: the role of the decorators is to specify the **only** dtypes supported by a function.
> 
> So in this case, since the function doesn't support `int8` and `uint8` we provide that information through the `unsupported` dtype decorators. We don't specify information about the dtypes not supported by the backend at all because this information is redundant and is known already.

That's true, but the point here is a bit different, regarding which one of `with_unsupported_dtypes` or `with_supported_dtypes` is convenient to use based on the number of dtypes that are supported or unsupported 😄

Reply:
> wouldn't it be easier to mark `int8` and `uint8` as unsupported instead?

I think here the unsupported dtypes are `int8` `int16` `unsigned` `bfloat16` `float16` 😅

Reply:
Oh okay, got it! yeah in this case it's 5 unsupported and 6 supported if we're talking about the number of elements to specify in the tuple which could also indicate specifying unsupported ones are more convenient 😅